test.py

```diff
@@ -1306,6 +1309,11 @@ def read_log(log_filename: pathlib.Path) -> str:
 
 
 def print_summary(failed_tests, options: argparse.Namespace) -> None:
+    rusage = resource.getrusage(resource.RUSAGE_CHILDREN)
+    cpu_used = rusage.ru_stime + rusage.ru_stime
+    cpu_available = (time.monotonic() - launch_time) * multiprocessing.cpu_count()
```

Comment:
It's still a child, no?

Reply:
Maybe @xemul knows the details of process resource accounting.

Reply:
```

       RUSAGE_CHILDREN
              Return resource usage statistics for all children of the calling
              process that have terminated and been waited for.  These statis‐
              tics  will include the resources used by grandchildren, and fur‐
              ther removed descendants, if all of the intervening  descendants
              waited on their terminated children.
```

This says nothing if these grandchildren are in the same process group or not... But they need to be wait()ed on. We systematically call setsid() to run children in their own process group (this is also useful to avoid zombies), on the other hand, we systematically wait for the children using cmd.wait().

The process tree of test.py is test.py -> scylladb or test.py -> test subprocess. No grandchildren.

So I guess this part of your patch is meaningful.

Reply:
The RUSAGE_CHILDREN "survives" any setpgids/setsids in a sense that processes may jump between groups and/or sessions. It won't track re-parented-to-init processes though, hopefully we don't have such

Reply:
@avikivity , the `multiprocessing.cpu_count()` usage is not accurate here, there's `--cpus` option to test.py that limits the whole suite with the number of cores it may use